apps/wallet-mobile/src/features/Swap/common/SwapProvider.tsx

```diff
@@ -0,0 +1,506 @@
+        throw new Error(`swapReducer invalid action`)
+    }
+  })
+}
+
+export const SwapAction = {
+  ChangeOrderType: 'ChangeOrderType',
+  TokenInInputTouched: 'TokenInInputTouched',
+  TokenOutInputTouched: 'TokenOutInputTouched',
+  TokenInIdChanged: 'TokenInIdChanged',
+  TokenOutIdChanged: 'TokenOutIdChanged',
+  TokenInAmountChanged: 'TokenInAmountChanged',
+  TokenOutAmountChanged: 'TokenOutAmountChanged',
+  TokenInErrorChanged: 'TokenInErrorChanged',
+  TokenOutErrorChanged: 'TokenOutErrorChanged',
+  WantedPriceInputChanged: 'WantedPriceInputChanged',
+  SlippageInputChanged: 'SlippageInputChanged',
+  SwitchTouched: 'SwitchTouched',
+  DexSelectorTouched: 'DexSelectorTouched',
+  Refresh: 'Refresh',
+  ResetAmounts: 'ResetAmounts',
+  ResetForm: 'ResetForm',
+  EstimateResponse: 'EstimateResponse',
+  EstimateError: 'EstimateError',
+  CreateResponse: 'CreateResponse',
+  CreateError: 'CreateError',
+} as const
+
+type SwapActionValueMap = {
+  ChangeOrderType: 'limit' | 'market'
+  TokenInInputTouched: undefined
+  TokenOutInputTouched: undefined
+  TokenInIdChanged: Portfolio.Token.Id
+  TokenOutIdChanged: Portfolio.Token.Id
+  TokenInAmountChanged: string
+  TokenOutAmountChanged: string
+  TokenInErrorChanged: string | null
+  TokenOutErrorChanged: string | null
+  WantedPriceInputChanged: string
+  SlippageInputChanged: number
+  SwitchTouched: undefined
+  DexSelectorTouched: Swap.Provider
+  Refresh: undefined
+  ResetAmounts: undefined
+  ResetForm: undefined
+  EstimateResponse: Swap.EstimateResponse
+  EstimateError: Api.ResponseError
+  CreateResponse: Swap.CreateResponse
+  CreateError: Api.ResponseError
+}
+
+export type SwapAction = {
+  [K in keyof SwapActionValueMap]: SwapActionValueMap[K] extends undefined
+    ? {type: K}
+    : {type: K; value: SwapActionValueMap[K]}
+}[keyof SwapActionValueMap]
+
+const defaultState: SwapState = Object.freeze({
+  reqres: 'response',
+  orderType: 'market',
+  lastInputTouched: 'in',
+  tokenInInput: {
+    isTouched: true,
+    tokenId: primaryTokenId,
+    disabled: false,
+    error: null,
+    value: '',
+  },
+  tokenOutInput: {
+    isTouched: false,
+    tokenId: undefined,
+    disabled: false,
+    error: null,
+    value: '',
+  },
+  slippageInput: {
+    value: 1,
+  },
+  selectedDex: {
+    isTouched: false,
+    value: undefined,
+  },
+  wantedPrice: '',
+  canSwap: false,
+  estimate: undefined,
+  createTx: undefined,
+  cancelTx: undefined,
+  cancelError: undefined,
+} as const)
+
+type SwapState = {
+  reqres: 'request' | 'response'
+  orderType: 'market' | 'limit'
+  lastInputTouched: 'in' | 'out'
+  tokenInInput: {
+    isTouched: boolean
+    tokenId?: Portfolio.Token.Id
+    disabled: boolean
+    error: string | null
+    value: string
+  }
+  tokenOutInput: {
+    isTouched: boolean
+    tokenId?: Portfolio.Token.Id
+    disabled: boolean
+    error: string | null
+    value: string
+  }
+  slippageInput: {
+    value: number
+  }
+  selectedDex: {
```

Comment:
`selectedDex` or Liquidity Pool/Provider?

Reply:
https://emurgo.slack.com/archives/GCN6JLZK7/p1738149957332599, lets see the outcome.

Reply:
It gets weird, because you can list the liquidity providers and for example get 2 pools on the same dex. But when the time comes to create the swap, you can only specify which dex to use, not which pool.